src/app/views/canvas/snapping.tsx

```diff
@@ -325,7 +326,24 @@ export class MarkSnappingSession extends SnappingSession<
     bound: Prototypes.Handles.Description,
     threshold: number
   ) {
-    super(guides.filter(x => x.element != element), bound, threshold);
+    super(
+      guides.filter(x => {
+        // element cannot snap to itself
+        if (x.element === element) {
+          return false;
+        }
+        // special rules for guides
+        if (element.classID === GuideClass.classID) {
```

Comment:
Charticulator has function `isType` for type checking https://github.com/microsoft/charticulator/blob/aa3f4135069bfb4c5e88e8ff77ee680327943246/src/core/prototypes/object.ts#L184

Reply:
Thanks! Fixed